### PR TITLE
feat: add presence helpers to StateManager

### LINKED ISSUE #1068: Add presence helper methods to Api (anyone_home, everyone_home, noone_home)
## Description

Presence checks are among the most common conditions in home automation. AppDaemon ships them as first-class helpers:

```python
if self.anyone_home():
    ...
if self.everyone_home(person='jessica'):
    ...
if self.noone_home():
    ...
```

In Hassette today you'd query `device_tracker` or `person` entities manually through `self.states` — possible, but non-obvious enough that every app author has to figure out the entity structure themselves and write the same aggregation logic.

These helpers belong on `Api` (or `App` as a thin delegator) since presence is a read against HA state, not a bus subscription. They'd wrap the `person.*` domain (preferred over `device_tracker` directly, which tracks individual devices rather than people).

## Acceptance Criteria

- [ ] `Api.anyone_home(person: str | None = None) -> bool` — `True` if at least one person (or the specified person) is home
- [ ] `Api.everyone_home(person: str | None = None) -> bool` — `True` if all tracked persons (or the specified person) are home
- [ ] `Api.noone_home() -> bool` — `True` if no tracked persons are home
- [ ] Implementation queries `person.*` entities; falls back to `device_tracker.*` if no `person` entities are configured
- [ ] Exposed on `App` instances as convenience delegators (same pattern as `call_service`, `get_state`, etc.)
- [ ] Composable with the constraint system from #1066 (e.g. `P.noone_home()` predicate factory, or documented pattern for use with `where=`)

### PR DESCRIPTION
**Pull request by NodeJSmith**
Adds synchronous, cache-backed presence helpers to `StateManager`, accessed via `self.states`. Closes the core of #1068; the predicate-composability criterion is split into #1129 so this lands cleanly.

### Presence helpers

Four methods on `StateManager` — each reads the local state cache, so there's no `await` and no network round-trip:

- `anybody_home()` — at least one tracked person is home
- `everybody_home()` — every tracked person is home (`False` when none are tracked)
- `nobody_home()` — no tracked person is home
- `is_home(entity_id)` — a single `person.*` / `device_tracker.*` entity is home

They read the `person` domain, falling back to `device_tracker` only when no `person` entities exist. `"home"` is the value both domains report in the home zone; anything else (`"not_home"`, a named zone) counts as away.

### Why StateManager, not Api or App

Presence is a synchronous read of *current* state — exactly what `StateManager` is for. Building it on `Api.get_states()` (the issue's suggestion) would do a full WebSocket fetch of every entity on each check; `self.states.person` already holds the answer in the local cache, synchronously.

App-level delegators were intentionally **not** added. The issue's premise ("same pattern as `call_service`/`get_state`") was inaccurate — those live on `self.api`, not `App` — and `App` deliberately keeps a minimal surface (only `now()` today).

### API design decisions

- **Dropped the issue's `person=` parameter** on the quantifiers. With a single person specified, "anybody" and "everybody" collapse into the same question, so a dedicated `is_home(entity_id)` covers the singular case instead of two methods that do the same thing.
- **Method names use `-body`** (`anybody`/`everybody`/`nobody`) rather than AppDaemon's `anyone`/`everyone`/`noone` — uniform and cleaner to read (`noone` has no clean single-word form).

### Tests & docs

- 13 unit tests in `tests/unit/test_state_manager.py` (quantifiers, `is_home`, `device_tracker` fallback, person-domain preference, no-entities edge case)
- New "Presence" section + a Pyright-checked snippet under `docs/pages/core-concepts/states/`
- `.pyi` stub kept in sync (it shadows the `.py` for type-checkers)

No frontend or schema changes — these are pure read helpers.

Follow-up: #1129 (presence as `P.*` predicate factories for `where=` filters).

Closes #1068

### CODE REVIEW

File: docs/pages/core-concepts/states/snippets/states_presence.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from hassette import App
+
+
+class PresenceApp(App):
+    async def on_initialize(self):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply docs/pages/core-concepts/states/snippets/states_presence.py:5]: Declining this one. Across the docs snippets, `on_initialize` omits the `-> None` return annotation in 123 of 174 occurrences — the unannotated form is the established convention here, and this snippet matches it. Adding the annotation to just this one would make it inconsistent with its neighbors. A repo-wide `-> None` sweep on lifecycle hooks in snippets would be a separate change, not a one-liner on this PR.
